src/java.desktop/share/classes/javax/swing/text/html/StyleSheet.java

```diff
@@ -1014,12 +1014,28 @@ public float getPointSize(String size) {
     }
 
     /**
-     * Converts a color string such as "RED" or "#NNNNNN" to a Color.
-     * Note: This will only convert the HTML3.2 color strings
-     *       or a string of length 7;
-     *       otherwise, it will return null.
+     * Converts a color string such as "RED",  "rgb(r g b)",
+     * "rgba(r g b a)" or "#NNN", "#NNNN", "#NNNNNN",
+     * "#NNNNNNNN" to a Color.
```

Comment:
This specification change seems to be dependent on the change in implementation proposed in
https://github.com/openjdk/jdk/pull/10317/
JDK-8293776 : Adds CSS 4 and 8 digits hex coded Color
And BTW if you implemented that change on its own, it would not be spec compliant.

I think it best to WITHDRAW that other PR and absorb it in this PR

Also you should in the doc include the spec link.
spec limits what it will convert. I think

Reply:
Initially, it looked as if handling these issues separately was a good idea. Now that we know that the specification of a public needs updating, which requires a CSR, it is better to have everything ready: named colors, hex-parsers, rgb() and rgba() parsers. The spec refers to all the cases. If one is missing, the method is not compliant to its specification.

Amending the implementation one by one would require updating the spec and reviewing the CSR. Thus, having everything ready in one larger PR makes more sense.